cmd/fleet/cron.go

```diff
@@ -745,11 +751,68 @@ func newCleanupsAndAggregationSchedule(
 				return ds.UpdateOSVersions(ctx)
 			},
 		),
+		schedule.WithJob(
+			"verify_disk_encryption_keys",
+			func(ctx context.Context) error {
+				return verifyDiskEncryptionKeys(ctx, logger, ds, config)
+			},
+		),
 	)
 
 	return s, nil
 }
 
+func verifyDiskEncryptionKeys(
+	ctx context.Context,
+	logger kitlog.Logger,
+	ds fleet.Datastore,
+	config *config.FleetConfig,
+) error {
+	if !config.MDM.IsAppleSCEPSet() {
+		logger.Log("inf", "skipping verification of encryption keys as MDM is not fully configured")
+		return nil
+	}
+
+	keys, err := ds.GetUnverifiedDiskEncryptionKeys(ctx)
+	if err != nil {
+		logger.Log("err", "unable to get unverified disk encryption keys from the database", "details", err)
+		return err
+	}
+
+	cert, _, _, err := config.MDM.AppleSCEP()
+	if err != nil {
+		logger.Log("err", "unable to get SCEP keypair to decrypt keys", "details", err)
+		return err
+	}
+
+	parsed, err := x509.ParseCertificate(cert.Certificate[0])
+	if err != nil {
+		logger.Log("err", "unable to get SCEP keypair to decrypt keys", "details", err)
+		return err
+	}
```

Comment:
If we actually need the certificate leaf, we should do it from the start in `config.MDM.AppleSCEP`, the same way that we do it for `AppleAPNs` (look at the call to `pair.Parse(true)`). That way, you could remove this whole `ParseCertificate` block and use `cert.Leaf` instead, it would be filled at fleet startup (like it is for APNs). We didn't need the leaf when I implemented that `AppleSCEP` call which is why I didn't do it for SCEP.

Reply:
for a moment while writing this bit I was on the fence, but I didn't want to change the default for whatever reason. Updated, thank you!!